app/controllers/WhichSubscriptionController.scala

```diff
@@ -0,0 +1,91 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import controllers.routes
+import forms.WhichSubscriptionFormProvider
+import javax.inject.Inject
+import models.Mode
+import navigation.Navigator
+import pages.WhichSubscriptionPage
+import play.api.Logger
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import services.ProfessionalBodiesService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.WhichSubscriptionView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class WhichSubscriptionController @Inject()(
+                                             override val messagesApi: MessagesApi,
+                                             sessionRepository: SessionRepository,
+                                             navigator: Navigator,
+                                             identify: IdentifierAction,
+                                             getData: DataRetrievalAction,
+                                             requireData: DataRequiredAction,
+                                             formProvider: WhichSubscriptionFormProvider,
+                                             val controllerComponents: MessagesControllerComponents,
+                                             view: WhichSubscriptionView,
+                                             professionalBodiesService: ProfessionalBodiesService
+                                           )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      val preparedForm = request.userAnswers.get(WhichSubscriptionPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      professionalBodiesService.localSubscriptions().map(
+        subscriptions =>
+          Ok(view(preparedForm, mode, subscriptions))
+      ).recoverWith{
+        case e =>
+          Logger.error("failed to load subscriptions", e)
+          Future.successful(Redirect(routes.TechnicalDifficultiesController.onPageLoad()))
+      }
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      professionalBodiesService.localSubscriptions().flatMap {
```

Comment:
Do you want to be pulling this data down twice? Is it possible/wise to cache it?